File: template/nest-next/src/auth/auth.controller.ts
import { Body, Controller, Post } from '@nestjs/common';
import { AuthService } from './auth.service';
import {
  CreateApiTokenDto,
  LoginDto,
  LogoutAuthDto,
  RefreshTokenDTO,
  RevokeApiTokenDto,
} from './dto';
import { ApiCreatedResponse, ApiOperation } from '@nestjs/swagger';
import { ApiToken, TokenPair } from './entity';
import { ApiTokenService } from './api-token.service';
import { Public } from './deocrators';

@Controller('auth')
export class AuthController {
  constructor(
    private readonly authService: AuthService,
    private readonly apiTokenService: ApiTokenService,
  ) {}

  @ApiOperation({
    summary: '刷新令牌对',
    description: '刷新令牌对, 返回一个新的令牌对',
  })
  @ApiCreatedResponse({
    type: TokenPair,
  })
  @Public()
  @Post('/token/refresh')
  async refreshToken(@Body() body: RefreshTokenDTO) {
    return this.authService.refreshToken(body.token);
  }

  @ApiOperation({
    summary: '登录',
  })
  @ApiCreatedResponse({
    type: TokenPair,
  })
  @Public()
  @Post('login')
  async login(@Body() body: LoginDto) {
    return this.authService.login(body);
  }

  @ApiOperation({
    summary: '登出',
  })
  @Post('logout')
  logout(@Body() body: LogoutAuthDto) {
    return this.authService.logout(body.token);
  }

  @ApiOperation({
    summary: '生成API Token',
    description: '用于外部系统调用',
  })
  @ApiCreatedResponse({
    type: ApiToken,
  })
  // 生成API Token，用于外部系统调用
  @Public()
  @Post('api-token')
  async generateApiToken(@Body() body: CreateApiTokenDto) {
    return this.apiTokenService.issueApiToken(body);
  }

  @ApiOperation({
    summary: '撤销API Token',
    description: '用于外部系统调用',
  })
  @Post('revoke-api-token')
  async revokeApiToken(@Body() body: RevokeApiTokenDto) {
    return this.apiTokenService.revokeApiToken(body.email, body.tokenId);
  }
}

File: template/springboot/src/main/java/com/TinyPro/controller/AuthController.java
package com.TinyPro.controller;

import com.TinyPro.annotation.IsPublic;
import com.TinyPro.entity.dto.*;
import com.TinyPro.entity.vo.ApiToken;
import com.TinyPro.entity.vo.MenuTreeVo;
import com.TinyPro.entity.vo.TokenPair;
import com.TinyPro.service.IAuthService;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.media.Content;
import io.swagger.v3.oas.annotations.media.Schema;
import io.swagger.v3.oas.annotations.responses.ApiResponse;
import io.swagger.v3.oas.annotations.responses.ApiResponses;
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import jakarta.validation.Valid;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.Map;

@RestController
@RequestMapping("/auth")
public class AuthController {
    @Autowired
    private IAuthService authService;

    @IsPublic()
    @PostMapping("/token/refresh")
    public ResponseEntity<TokenPair> refreshToken(@RequestBody @Valid RefreshTokenDto refreshTokenDto) {
        return ResponseEntity.ok(authService.refreshToken(refreshTokenDto.getToken()));
    }

    @IsPublic()
    @PostMapping("/api-token")
    @Operation(summary = "生成 API Token", description = "使用邮箱和密码生成供外部系统调用的 API Token")
    @ApiResponses({
            @ApiResponse(responseCode = "200", description = "生成成功"),
            @ApiResponse(responseCode = "400", description = "账号或密码错误"),
            @ApiResponse(responseCode = "404", description = "用户不存在")
    })
    public ResponseEntity<ApiToken> generateApiToken(@RequestBody @Valid CreateApiTokenDto dto) {
        return ResponseEntity.ok(authService.generateApiToken(dto));
    }

    @IsPublic()
    @PostMapping("/login")
    public ResponseEntity<?> login(@RequestBody @Valid CreateAuthDto createAuthDto, HttpServletResponse response) throws Exception {
        return authService.login(createAuthDto, response);
    }

    @PostMapping("/logout")
    public ResponseEntity<Map<String, Boolean>> logout(
            @Valid @RequestBody LogoutAuthDto logoutAuthDto,
            HttpServletRequest request
    ) {
        authService.logout(extractBearerToken(request));
        return ResponseEntity.ok(Map.of("success", true));
    }

    @PostMapping("/revoke-api-token")
    @Operation(summary = "撤销 API Token", description = "按邮箱和 tokenId 撤销指定 API Token")
    @ApiResponses({
            @ApiResponse(responseCode = "201", description = "撤销成功"),
            @ApiResponse(responseCode = "401", description = "未认证", content = @Content(schema = @Schema(hidden = true)))
    })
    public ResponseEntity<String> revokeApiToken(@RequestBody @Valid RevokeApiTokenDto dto) {
        authService.revokeApiToken(dto);
        return ResponseEntity.status(HttpStatus.CREATED).build();
    }

    @GetMapping("/test")
    public ResponseEntity<MenuTreeVo> gettext() {
        return ResponseEntity.ok(new MenuTreeVo());
    }

    private String extractBearerToken(HttpServletRequest request) {
        String header = request.getHeader("Authorization");
        if (header == null || !header.startsWith("Bearer ")) {
            return null;
        }
        return header.substring(7);
    }
}
